### ISSUE TITLE
Migrating actors in sharding takes a long time after killing one node (in 2.3.8+).

### ISSUE THREAD

**Issue by otrebski** [akka/akka#18101]
Hi.
I have played with sharding.  My scenario is to run a few JVM's. I'm killing one of the JVM. In akka up to 2.3.7, sharding needs ~4 seconds to migrate actors to the other node (when cluster node is unreachable).
If I'm using akka 2.3.8+ or 2.4+, it takes 40 seconds. After ~4 seconds node is unreachable, after ~24 seconds node is down, after ~40 seconds actor is migrated.

I'm sending every 100ms message to shared actors.

My configuration is:
```hacon

akka {
  loggers = ["akka.event.slf4j.Slf4jLogger"]
  loglevel = ERROR

  actor {
    provider = "akka.cluster.ClusterActorRefProvider"
  }

  remote {
    enabled-transports = [akka.remote.netty.tcp]
    log-remote-lifecycle-events = off
    netty.tcp {
      hostname = ${HOST}
      port = ${PORT}
    }
  }

  cluster {
    seed-nodes = [
      ${SEED_NODE}
    ]
    auto-down-unreachable-after = 20s
  }
  persistence {
    journal.plugin = "akka.persistence.journal.leveldb-shared"
    journal.leveldb-shared.store {
      # DO NOT USE 'native = off' IN PRODUCTION !!!
      native = off
      dir = "target/shared-journal"
    }
    snapshot-store.local.dir = "target/snapshots"
  }
}
```

**Comment by patriknw** [comment]
thanks for reporting, I'll check

**Comment by otrebski** [comment]
I have encounter this issue when I was creating project: https://github.com/otrebski/reactive-missile-defend

Simple demonstration of issue you can find at: https://github.com/otrebski/akka-issue18101/tree/master

**Comment by patriknw** [comment]
Thanks a lot, I have not been able to look at this yet.

For 2.4.0-RC1 you can try with config

    akka.cluster.down-removal-margin=10ms

That is something we will disable by default in RC2, https://github.com/akka/akka/pull/18338

The 24 seconds before down is due to

    auto-down-unreachable-after = 20s

**Comment by patriknw** [comment]
@otrebski I have tested the scenario with 2.4.0-RC2 and it works as expected. The crashed (unreachable) node must be downed and thereafter removed before the shard region can be migrated to another node. Same with the shard coordinator. 

The shorter time in 2.3.7 was due to a bug.

One way to speed up fail over when you have a controlled shutdown is to use the Cluster `leave` command. Then the downing is not needed and is therefore much faster. This can be emitted from any node with `Cluster(system).leave` or via JMX.
